monGARS/core/cortex/curiosity_engine.py

```
@@ -1,114 +1,213 @@
+            logger.debug("Vector similarity fallback due to invalid session factory")
+            return []
+        try:
+            async with session_factory as session:
                 result = await session.execute(
                     select(ConversationHistory.query)
                     .order_by(ConversationHistory.timestamp.desc())
-                    .limit(25)
+                    .limit(limit)
                 )
-                history: Iterable[str] = [row[0] for row in result if row[0]]
         except Exception as exc:  # pragma: no cover - DB optional
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)